controllers/vgmanager_daemonset.go

```diff
@@ -112,17 +117,30 @@ var (
 )
 
 // newVGManagerDaemonset returns the desired vgmanager daemonset for a given LVMCluster
-func newVGManagerDaemonset(lvmCluster lvmv1alpha1.LVMCluster) appsv1.DaemonSet {
+func newVGManagerDaemonset(r *LVMClusterReconciler, ctx context.Context, lvmCluster *lvmv1alpha1.LVMCluster) appsv1.DaemonSet {
 	// aggregate nodeSelector and tolerations from all deviceClasses
 	nodeSelector, tolerations := extractNodeSelectorAndTolerations(lvmCluster)
 	volumes := []corev1.Volume{LVMDConfVol, DevHostDirVol, UDevHostDirVol, SysHostDirVol}
 	volumeMounts := []corev1.VolumeMount{LVMDConfVolMount, DevHostDirVolMount, UDevHostDirVolMount, SysHostDirVolMount}
 	privileged := true
 	var zero int64 = 0
+
+	// try to get running pod image, on failure fallback to default image
+	image, err := getRunningPodImage(r, ctx)
+	if err != nil {
+		r.Log.Error(err, "failed to get image from running operator")
+		image = VGManagerImage
```

Comment:
IMO, we should either use `env` or use `getRunningPodImage`. We don't need both.

Reply:
and this would also mean we would never be able to test the custom vgmanager image in the `env`. Since it will always use the `lvm-operator` image!